test/unit/plugins/divclass.test.js

```diff
@@ -117,5 +140,48 @@ describe('divclass', () => {
       const output = parser.sourceAndRedactedToHtml(source, redacted);
       expect(output).toEqual("<div class=\"complex-example\"><ul>\n<li>une liste ordonnée</li>\n<li>avec <a href=\"http://example.com\">d'autres contenus rédigés</a></li>\n</ul></div>\n");
     });
+
+    it('can restore content with reordered indexes', () => {
+      const source = "[zero]\n\nzero\n\n[/zero]\n\n[one]\n\none\n\n[/one]";
+      const redacted = "[][1]\n\nun\n\n[/][1]\n\n[][0]\n\nzéro\n\n[/][0]";
+      const output = parser.sourceAndRedactedToHtml(source, redacted);
+      expect(output).toEqual("<div class=\"one\"><p>un</p></div>\n<div class=\"zero\"><p>zéro</p></div>\n");
+    });
+
+    it('can restore content with nesting changes', () => {
+      const source = "[zero]\n\n[one]\n\n\n\n[/one]\n\n[/zero]"
+
+      const unNestedRedaction = "[][0]\n\n\n\n[/][0]\n\n[][1]\n\n\n\n[/][1]";
+      expect(parser.sourceAndRedactedToHtml(source, unNestedRedaction))
+        .toEqual("<div class=\"zero\"></div>\n<div class=\"one\"></div>\n");
+
+      const invertedRedaction = "[][1]\n\n[][0]\n\n\n\n[/][0]\n\n[/][1]";
+      expect(parser.sourceAndRedactedToHtml(source, invertedRedaction))
+        .toEqual("<div class=\"one\"><div class=\"zero\"></div></div>\n");
+
+      const brokenRedaction = "[][0]\n\n[][1]\n\n\n\n[/][0]\n\n[/][1]"
+      expect(parser.sourceAndRedactedToHtml(source, brokenRedaction))
+        .toEqual("<div class=\"zero\"><p><a href=\"\"></a></p></div>\n<p><a href=\"\">/</a></p>\n");
```

Comment:
Fascinating! This feels like a good consistent way to fail, with the downside that the empty links render invisibly; not sure I'd recommend anything else though.